docs: add vscode develoment section (#5756)
renovatebot/renovate: docs/development/local-development.md

  OpenJDK 64-Bit Server VM 18.9 (build 11.0.6+10, mixed mode)
  ```

_VSCode_

If you have [VSCode](https://code.visualstudio.com/) and [Docker](https://www.docker.com/get-started) or [WSL](https://docs.microsoft.com/en-us/windows/wsl) (Windows only) installed, you can make use of [remote development feature](https://code.visualstudio.com/docs/remote/remote-overview) of VSCode. Then you can skip installing dev tools locally.

- Install [git](https://git-scm.com/downloads), make sure you've [configured your username and email](https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup)

> - Install [Remote - Containers](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) extension and [requirements](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers#system-requirements). On Windows you can alternatively use WSL. For this you need [Remote - WSL](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-wsl) extension. To get it fully working, you need to install the Linuy dependencies above to the WSL distro.

Reviewer: ```suggestion
- Install the [Remote - Containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) and [check its system requirements](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers#system-requirements).
```

I'd not mention WSL here as it makes stuff more complicated.

Author: I use WSL, it's a lot faster than container :upside_down_face:

Reviewer 2: I'd also leave that up to the end user

> - Open source code folder with VSCode

Reviewer: ```suggestion
- Open the repository folder in VS Code
```

> - Reopen Project in Container (small little green button on lower left of VSCode)

Reviewer: ```suggestion
- Choose "Reopen in Container" via the command palette or the small button in the lower left corner. 
```

The button is not always green :)

> VSCode [integrated terminal](https://code.visualstudio.com/docs/editor/integrated-terminal) should be running in container to run additional commands below.

Reviewer: ```suggestion
VS Code [integrated terminal](https://code.visualstudio.com/docs/editor/integrated-terminal) is now running in the container and can be used to run additional commands.
```
